cmd/sonobuoy/app/images.go

```diff
@@ -101,3 +164,249 @@ func getImages(cmd *cobra.Command, args []string) {
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init upstream registry list"))
+			os.Exit(1)
+		}
+
+		ctx := context.Background()
+		cli, err := client.NewClientWithOpts(client.WithVersion(dockerClientVersion))
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init docker client"))
+			os.Exit(1)
+		}
+
+		for _, v := range upstreamImages {
+			err = image.PullImage(ctx, cli, v)
+			if err != nil {
+				errlog.LogError(errors.Wrapf(err, "couldn't pull image: %v", v.GetE2EImage()))
+			}
+			fmt.Println("########")
+		}
+	default:
+		errlog.LogError(errors.Errorf("Unsupported plugin: %v", imagesflags.plugin))
+		os.Exit(1)
+	}
+}
+
+func downloadImages(cmd *cobra.Command, args []string) {
+	switch imagesflags.plugin {
+	case "e2e":
+
+		cfg, err := imagesflags.kubeconfig.Get()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get REST client"))
+			os.Exit(1)
+		}
+
+		sbc, err := getSonobuoyClient(cfg)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "could not create sonobuoy client"))
+			os.Exit(1)
+		}
+
+		version, err := sbc.Version()
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't get Sonobuoy client"))
+			os.Exit(1)
+		}
+
+		upstreamImages, err := image.GetImages("", version)
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init upstream registry list"))
+			os.Exit(1)
+		}
+
+		ctx := context.Background()
+		cli, err := client.NewClientWithOpts(client.WithVersion(dockerClientVersion))
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't init docker client"))
+			os.Exit(1)
+		}
+
+		images := []string{}
+		for _, v := range upstreamImages {
+			images = append(images, v.GetE2EImage())
+		}
+		err = image.SaveToTar(ctx, cli, images, image.GetTarFileName(version))
+		if err != nil {
+			errlog.LogError(errors.Wrap(err, "couldn't save images to tar"))
+			os.Exit(1)
+		}
+
+	default:
+		errlog.LogError(errors.Errorf("Unsupported plugin: %v", imagesflags.plugin))
+		os.Exit(1)
+	}
+}
+
+func pushImages(cmd *cobra.Command, args []string) {
+
+	switch imagesflags.plugin {
+	case "e2e":
+
+		// Check if the e2e file exists
+		if _, err := os.Stat(imagesflags.e2eRegistryConfig); os.IsNotExist(err) {
+			errlog.LogError(errors.Errorf("file does not exist or cannot be opened: %v", imagesflags.e2eRegistryConfig))
+			os.Exit(1)
+		}
+
+		// Check if username if specified for registry auth
+		if len(imagesflags.username) > 0 {
+			// Check if password was set via ENV variable, otherwise prompt user for password from STDIN
+			envPassword := viper.GetString("password")
+			if len(envPassword) == 0 {
+				fmt.Print("Registry password: ")
+				bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+				if err != nil {
+					errlog.LogError(errors.Wrap(err, "couldn't get password from user"))
+					os.Exit(1)
+				}
+
+				imagesflags.password = string(bytePassword)
+				fmt.Print("\n")
+			} else {
+				imagesflags.password = envPassword
```

Comment:
nit: just move this up to the line grabbing it with `viper` so you can remove the entire `else` block/path.

Reply:
What if it's not passed via env and I want to prompt the user?